lib/Widgets/Avatar.vala

```diff
@@ -28,10 +28,34 @@ public class Granite.Widgets.Avatar : Gtk.EventBox {
     private const string DEFAULT_STYLE = "avatar";
     private const int EXTRA_MARGIN = 4;
     private bool draw_theme_background = true;
+    private string? filename;
 
     public Gdk.Pixbuf? pixbuf { get; set; }
 
     /**
+     * The "pixel-size" property can be used to specify a fixed size.
+     */
+    public int pixel_size { get; construct; }
+
+    /**
+     * The "file" property can be used to set a new avatar image from a file path.
+     */
+    public string? file {
```

Comment:
I used `file` because that's how `Gtk.Image` does it: https://valadoc.org/gtk+-3.0/Gtk.Image.file.html . Trying to emulate Gtk+ API here as much as possible. It also matches the name of the creation method since it is `from_file` and not `from_filename`. Same as `Gtk.Image.from_file` which also takes a file: https://valadoc.org/gtk+-3.0/Gtk.Image.Image.from_file.html

Yeah you're right, I should check for null and not rely on the catch for that. will fix!